pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+
+    ###################################################################
+    #
+    #     General support methods for the fake_... methods
+    #
+    #####################################################################
+
+    def _class_exists(self, namespace, cl):
+        """
+        Test if class exists in repository namespace
+        """
+        if namespace in self.classes:
+            if self.classes[namespace][cl]:
+                return True
+        return False
+
+    @staticmethod
+    def _make_tuple(rtn_value):
+        """
+        Make the return value into a tuple in accord with _imethodcall
+        """
+        return [("IRETURNVALUE", {}, rtn_value)]
+
+    @staticmethod
+    def _remove_qualifiers(obj):
+        """
+        Remove all qualifiers from the input object.  The object may
+        be an CIMInstance or CIMClass. Removes qualifiers from the object and
+        from properties, methods, and parameters
+
+        This is used to process the IncludeQualifier parameter for classes
+        and instances
+        """
+        assert isinstance(obj, (CIMInstance, CIMClass))
+        obj.qualifiers = NocaseDict()
+        for prop in obj.properties:
+            obj.properties[prop].qualifiers = NocaseDict()
+        if isinstance(obj, CIMClass):
+            for method in obj.methods:
+                obj.methods[method].qualifiers = NocaseDict()
+                for param in obj.methods[method].parameters:
+                    obj.methods[method].parameters[param].qualifiers = {}
+
+    @staticmethod
+    def _remove_classorigin(obj):
+        """
+        Remove all ClassOrigin attributes from the input object. The object
+        may be a CIMInstance or CIMClass.
+
+        Used to process the IncludeClassOrigin parameter of requests
+        """
+        assert isinstance(obj, (CIMInstance, CIMClass))
+        for prop in obj.properties:
+            obj.properties[prop].class_origin = None
+        if isinstance(obj, CIMClass):
+            for method in obj.methods:
+                obj.methods[method].class_origin = None
+
+    @staticmethod
+    def cln(cl_list):
+        """get list of classnames from list of classes"""
+        return [cl.classname for cl in cl_list]
+
+    def _validate_classes_namespace(self, namespace):
+        """
+        Validate that namespace exists in the class repository
+        """
+        if namespace not in self.classes:
+            raise CIMError(CIM_ERR_INVALID_NAMESPACE,
+                           'Namespace %s not found for classes' % namespace)
+        return
+
+    def _validate_instances_namespace(self, namespace):
+        """
+        Validate that namespace exists in the class repository
+        """
+        if namespace not in self.instances:
+            raise CIMError(CIM_ERR_INVALID_NAMESPACE,
+                           'Namespace %s not found for instances' % namespace)
+        return
```

Comment:
Is that important to distinguish classes and instances when validating the namespace? It seems to me that could be merged into one method.